spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkChangelogScan.java

```diff
@@ -103,8 +104,7 @@ public StructType readSchema() {
 
   @Override
   public Batch toBatch() {
-    return new SparkChangelogBatch(
-        spark, table, readConf, taskGroups(), expectedSchema, hashCode());
+    return new SparkBatch(sparkContext, table, readConf, taskGroups(), expectedSchema, hashCode());
```

Comment:
If you take a look at `SparkChangelogBatch`, it was identical to `SparkBatch`, except the reader factory. Since we are adding a common factory for all tasks, it seemed appropriate to always leverage `SparkBatch`.